server/plugin.go

```diff
@@ -113,13 +120,42 @@ func (p *Plugin) OnActivate() error {
 		return mutateSampleLabels(l, target, *p.configuration.HonorTimestamps, []*relabel.Config{})
 	}
 
+	ticker := time.Tick(time.Duration(scrapeInterval) * time.Second)
+	p.closeChan = make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-p.closeChan:
+				return
+			case <-ticker:
+				cfg, err := p.configuration.Clone()
+				if err != nil {
+					_ = level.Error(logger).Log("msg", "configuration could not be cloned", "err", err)
+					break
+				}
+				buf := new(bytes.Buffer)
```

Comment:
Would suggest to move the creation outside the loop. And then reset the buffer (`buf.Reset`) for every scrape. This way we reduce GC.